JITImpl/LLVM/CMakeLists.txt

```diff
@@ -84,21 +84,38 @@ IF(LLVM_FOUND)
         src/LLVMJIT.cpp
     )
     ETISSPlugin(${PROJECT_NAME})
-    TARGET_LINK_LIBRARIES(${PROJECT_NAME} PUBLIC
-        -Wl,--start-group
-        ${LLVM_LIBS}
-        clangFrontend
-        clangCodeGen
-        clangLex
-        clangSerialization
-        clangAST
-        clangBasic
-        clangSema
-        clangEdit
-        clangAnalysis
-        clangDriver
-        clangParse
-        -Wl,--end-group
-    )
+    if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+        TARGET_LINK_LIBRARIES(${PROJECT_NAME} PUBLIC
```

Comment:
This should also work for the other platforms. The old was just a lazy way to avoid having to correctly order the libraries.

Reply:
I tried that. But Linux was failing to integrate the library. Please check the Linux tests here:
[https://github.com/AnoushkaU/etiss/runs/1523561952?check_suite_focus=true]

Reply:
Is that the correct link? It shows a successful linux build for me

Reply:
Yes, the build is successful, but if you check line 51 in Test(Linux/Mac) you will see the problem. It hence uses the default GCCJIT instead and gives a successful build.

Reply:
Hmm, maybe the order is not quite correct yet. The following could be very related:

https://github.com/pocl/pocl/issues/655

If you cannot make it work with little effort, leaving it like this is fine. In that case please just add a comment that the fixed order does not work on Linux.

Reply:
It works now after changing the order a bit like mentioned in the issue you highlighted!